cni/network/invoker_azure.go

```diff
@@ -48,6 +53,10 @@ func (invoker *AzureIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, er
 
 	// Call into IPAM plugin to allocate an address pool for the network.
 	addResult.ipv4Result, err = invoker.plugin.DelegateAdd(addConfig.nwCfg.Ipam.Type, addConfig.nwCfg)
+
+	if err != nil && strings.Contains(err.Error(), ipam.ErrNoAvailableAddressPools.Error()) {
```

Comment:
is there an issue with doing this via `errors.Is/As`?

Reply:
yes this error is from ipam and the ipam code unwrapping to string before writing to stdout and in cni side Error.Is couldn't find it

Reply:
From CNI specification: https://www.cni.dev/docs/spec/#Error
CNI delegate actions will type erase the error to a JSON string, so `errors.Is/As` won't work. Only way to check error type is doing it inside IPAM. Outside delegated functions, I can use string contains, or parse into a JSON and also use string contains for the Msg field.